rca/programmes/models.py

```diff
@@ -410,6 +448,10 @@ def clean(self):
             errors["curriculum_video"].append(
```

Comment:
This should be more reliable.

```
from wagtail.embeds import embeds
from wagtail.embeds.exceptions import EmbedException

try:
    embed = embeds.get_embed(self.curriculum_video)
except EmbedException:
    errors["curriculum_video"].append('invalid embed URL')
else:
    # not sure if this metadata is saved.
    if embed.provider_name.lower() !== 'youtube':
        # Only YouTube videos are supported for this field
```

https://github.com/wagtail/wagtail/blob/d6e4072e26ab7291d2aa56c4e87c0d82c7fb9030/wagtail/embeds/models.py#L29